src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -108,6 +114,10 @@ public class ExtensionsRunner {
      * A thread pool for the extension.
      */
     private final ThreadPool threadPool;
+    /**
+     * A task manager for the extension
+     */
+    private final TaskManager taskManager;
```

Comment:
Do we have a getter for TaskManager?

Reply:
@owaiskazi19 I tend to match variable names to their class names. It may be that later we need an `SDKTaskManager` class.

I had to include this class in injection as it is required to inherit from OpenSearch's `TransportAction` class to keep that same API, even though it's mostly null in our usage so far.  

We don't have a getter, the `TransportAction` constructor gets it via injection:
```java
@Inject
protected SampleTransportAction(String actionName, ActionFilters actionFilters, TaskManager taskManager) {
   ...
}
```

Alternatives are to create our own `TransportAction` superclass and update all the dynamic action code that @saratvemulapalli is working on.  I'd suggest if you have ideas on that you add them to #467 as there's still some improvement that can be done with this whole process. My goal was to unblock JS work that required getActions as a first step.